samples/tutorial/Tutorial3.md

```diff
@@ -0,0 +1,415 @@
+# Step 3
+
+_State throughout a tree of workflows_
+
+## Setup
+
+To follow this tutorial, launch Android Studio and open this folder (`samples/tutorial`).
+
+Start from the implementation of `tutorial-2-complete` if you're skipping ahead.
+
+## Editing todo items
+
+Now that a user can "log in" to their todo list, we want to add the ability to edit the todo items listed.
+
+### State ownership
+
+In the workflow framework, data flows _down_ the tree as properties to child workflows, and actions come _up_ as output events (as in the traditional computer science sense that trees that grow downward).
```

Comment:
"actions come _up_" seems a little confusing, since `WorkflowAction` and `action {}` actually mean something. 

Weird, it won't let me suggest changes. Perhaps:

In the workflow framework, data flows _down_ the tree as properties set by parents on their child workflows, and comes back _up_  from children as output events to their parents (as in the traditional computer science sense that trees that grow downward).

Reply:
Good point, this is a bit odd. This is directly copied from the swift tutorial, so might want to fix it there too (cc @AquaGeek )

Reply:
Done.